src/platform/WebAppWayland.cpp

```diff
@@ -124,18 +124,8 @@ void WebAppWayland::init(int width, int height, int surface_id)
 
     LOG_DEBUG("Width %d, Height %d", width, height);
 
-    setUiSize(width, height);
-    m_appWindow->InitWindow(width, height);
-
-    webos::WebAppWindowBase::LocationHint locationHint = getLocationHintFromString(m_locationHint);
-    if (locationHint != webos::WebAppWindowBase::LocationHint::kUnknown) {
-        m_appWindow->SetLocationHint(locationHint);
-    }
-
-    m_appWindow->setWebApp(this);
-
-    LOG_DEBUG("surface_ %p\n", surface_);
     if (surface_) {
+	    LOG_DEBUG("surface_ %p\n", surface_);
 	    switch (surface_->surface_type) {
```

Comment:
This should be eventually refactored to some other better place. We should not have here any AGL specific code.